app/scripts/metamask-controller.js

```diff
@@ -371,6 +372,10 @@ export default class MetamaskController extends EventEmitter {
     // the only thing that uses controller connections are open metamask UI instances
     this.activeControllerConnections = 0;
 
+    this.offscreenPromise = isManifestV3
```

Comment:
Could we initiate this sooner? This PR may end up delaying the initialization of the offscreen document by quite a lot, particularly on machines with slow disks (we load wallet state from disk before we get here).

It seems that your primary goal here was to have a Promise in `metamask-controller` that represents whether the offscreen document has loaded. We can still achieve that by passing it in as an option, by starting it earlier in `background.js` for example.

Reply:
Yeah, good point! I'll see if we can smoothly initiate it earlier

Reply:
Done!